index.js

```diff
@@ -1,22 +1,54 @@
+const { filterClosestPlaces, getRandomPlace, places } = require('./src/helpers');
+
 const Telegraf = require('telegraf');
 
 const bot = new Telegraf(process.env.BOT_TOKEN);
-/** @var {Array} */
-const places = require('./resources/places');
 
 bot.start((ctx) => {
     return ctx.reply('Working');
 });
 bot.command('/getplace', (ctx) => {
-    let countPlaces = places.length;
-    let randomPlaceId = Math.floor(Math.random() * countPlaces);
-
     /** @var {{title: String, lat: Number, lng: Number}} */
-    let randomPlace = places[randomPlaceId];
+    let randomPlace = getRandomPlace(places);
     ctx.replyWithMarkdown(`*Vieta:* ${randomPlace.title}
 *Lokācija:* ${randomPlace.lat}, ${randomPlace.lng}
     `).then(() => {
         return ctx.replyWithLocation(randomPlace.lat, randomPlace.lng)
     });
 });
+
+bot.command('/getplacel', (ctx) => {
+    let option = {
+        "parse_mode": "Markdown",
+        "reply_markup": {
+            "one_time_keyboard": true,
+            "keyboard": [[{
+                text: "Send my location",
+                request_location: true
+            }], ["Cancel"]]
+        }
+    };
+
+    ctx.replyWithMarkdown('Provide your location', option)
+    .then(() => {
+        bot.on("location", (ctx) => {
```

Comment:
šis izskatās, ka pēc pirmā trigger, piebindot on event locationam, katrs kurš sūtīs location saņems juanu place.
te droši vien atnāk message ar location...? jāpaskatās